### ISSUE TITLE
学习率应该怎么调试

### ISSUE THREAD

**Issue by zhangyuelong** [MaybeShewill-CV/lanenet-lane-detection#255]
请问应该怎么调节一下学习率，使其指数衰减，而不是这种，使用当前训练的网络可以测试图片和部分道路清晰视频效果勉强可以接受，但道路不清晰或车辆遮挡时效果差，请问应该怎么调试
 learning_rate = tf.train.exponential_decay(CFG.TRAIN.LEARNING_RATE, global_step,
                                                   100000, 0.1, staircase=True)

![2019-07-05 10-41-18屏幕截图](https://user-images.githubusercontent.com/20574313/60696210-5162bb80-9f17-11e9-8698-18c15d53bb15.png)

**Comment by zhangyuelong** [comment]
估计是因为staircase=True造成的

**Comment by MaybeShewill-CV** [comment]
@zhangyuelong staircase=False. 可以贴一些测试结果在这里如果可以的话

**Comment by zhangyuelong** [comment]
> @zhangyuelong staircase=False. 可以贴一些测试结果在这里如果可以的话
谢谢回复，我在试着，主要还是tf.train.exponential_decay参数没有设置对，稍后贴结果。exponential_decay(learning_rate, global_step, decay_steps, decay_rate, staircase=False, name=None)
参数：
    learning_rate - 初始学习率
    global_step - 用于衰减计算的全局步骤。 一定不为负数。喂入一次 BACTH_SIZE 计为一次 global_step
    decay_steps - 衰减速度，一定不能为负数，每间隔decay_steps次更新一次learning_rate值
    decay_rate - 衰减系数，衰减速率，其具体意义参看函数计算方程(对应α^t中的α)。
    staircase - 若 ‘ True ’ ，则学习率衰减呈 ‘ 离散间隔 ’ （discrete intervals），具体地讲，`global_step / decay_steps`是整数除法，衰减学习率（ the decayed learning rate ）遵循阶梯函数；若为 ’ False ‘ ，则更新学习率的值是一个连续的过程，每步都会更新学习率。

**Comment by zhangyuelong** [comment]
> @zhangyuelong staircase=False. 可以贴一些测试结果在这里如果可以的话

![2019-07-10 09-21-09屏幕截图](https://user-images.githubusercontent.com/20574313/60933275-2a7bff00-a2f4-11e9-9fec-e76a846f7cdb.png)
![2019-07-10 10-27-51屏幕截图](https://user-images.githubusercontent.com/20574313/60935864-d9710880-a2fd-11e9-9673-1a5629853434.png)

![2019-07-10 10-27-40屏幕截图](https://user-images.githubusercontent.com/20574313/60935836-c5c5a200-a2fd-11e9-85f9-e95bd95631ae.png)

![2019-07-10 10-19-08屏幕截图](https://user-images.githubusercontent.com/20574313/60935450-864a8600-a2fc-11e9-964c-79b34cb5c11b.png)
![2019-07-10 10-19-29屏幕截图](https://user-images.githubusercontent.com/20574313/60935611-07098200-a2fd-11e9-8e9a-678e3ced8006.png)

**Comment by MaybeShewill-CV** [comment]
@zhangyuelong １．你这里应该是出现了一定程度得过拟合 可能数据数据增强扩充样本．２．上图右出现两个车道线并线的情况是因为dbscan聚类错误导致， 这个你可以通过后处理的时候增加一些规则过滤这些错误， 或者调整dbscan参数:)

**Comment by zhangyuelong** [comment]
> @zhangyuelong １．你这里应该是出现了一定程度得过拟合 可能数据数据增强扩充样本．２．上图右出现两个车道线并线的情况是因为dbscan聚类错误导致， 这个你可以通过后处理的时候增加一些规则过滤这些错误， 或者调整dbscan参数:)
谢谢回复，训练数据确实不多，我在tusimple数据中挑选了1500张训练的，dbscan参数我也调试了，估计调试的不好，我在调试一下。

**Comment by zhangyuelong** [comment]
> @zhangyuelong １．你这里应该是出现了一定程度得过拟合 可能数据数据增强扩充样本．２．上图右出现两个车道线并线的情况是因为dbscan聚类错误导致， 这个你可以通过后处理的时候增加一些规则过滤这些错误， 或者调整dbscan参数:)

谢谢回复，训练数据确实不多，我在tusimple数据中挑选了1500张训练的，dbscan参数我也调试了，估计调试的不好，我在调试一下。

**Comment by MaybeShewill-CV** [comment]
@zhangyuelong 如果完全没有自己的训练数据可以考虑使用完整的tusimple数据集训练:)

**Comment by zhangyuelong** [comment]
> @zhangyuelong 如果完全没有自己的训练数据可以考虑使用完整的tusimple数据集训练:)
tusimple数据集中很多路面车道线是钉着原点，您觉对国内车道线检测是否有影响，影响大吗？，最近下了CULane数据集，不过CULane数据集也有很多乡村道路感觉使用lanenet网络处理不好，目前正在数据转化中．

**Comment by MaybeShewill-CV** [comment]
@zhangyuelong 和你具体使用的场景肯定有区别:)

**Comment by zhangyuelong** [comment]
> @zhangyuelong 和你具体使用的场景肯定有区别:)

好的，谢谢了，这个问题可以关了．
